### DISCUSSION TITLE
Access to `serverOptions` in eleventy config

### DISCUSSION THREAD

**Discussion by robole** [11ty/eleventy#4068] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

I created a `absoluteurl` filter that converts a relative URL or an absolute path to an absolute URL.  I would like it to behave like [ Jekyll's `absolute_url` filter](https://jekyllrb.com/docs/liquid/filters/). It prepends a `baseurl` value to the input.

The filter should do the following:
- `{{ "/about.html" | absoluteurl }}` would produce "https://localhost:8080/about.html" in dev. Whatever is the default of the local server.
- `{{ "/about.html" | absoluteurl }}` would produce "https://www.roboleary.net/about.html" in production when I provide a value of `https://www.roboleary.net` to `baseurl`.

Since I cannot obtain the server host name and port, I am unable to do this for dev mode. Instead I return the input `url`.

### Describe the solution you'd like

Access the `serverOptions` object with details about server.

### Describe alternatives you've considered

I tried to use `eleventyConfig.getMergingConfigObject().serverOptions` in the eleventy config file, however it is empty. I guess that this could be because the server is not initiated yet. Is this a timing issue?

### Additional context

This is the current implementation:

```javascript
function absoluteUrl(url, baseUrl, production = true) {
  if (baseUrl === undefined || baseUrl === null) {
    throw new Error(
      `absoluteUrl - The baseUrl parameter is missing.`
    );
  }

  let absUrl = url;

  if (production) {
    try {
      absUrl = new URL(url, baseUrl).toString();
    } catch (e) {
      debug(
        "Trying to convert %o to be an absolute url with baseUrl %o and failed, returning: %o (invalid url)",
        url,
        baseUrl,
        url
      );
    }
  }

  return absUrl;
}

**Comment by zachleat** [comment]
A bit of internal complexity here as the dev server currently starts after the build.

> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
